apps/dsa.c

```diff
@@ -8,7 +8,7 @@
  */
 
 #include <openssl/opensslconf.h>
-#ifdef OPENSSL_NO_DSA
+#if defined(OPENSSL_NO_DSA) || defined(OPENSSL_NO_DEPRECATED_3_0)
```

Comment:
This file should get an added `#define OPENSSL_SUPPRESS_DEPRECATED`, just like dhparam.c, and the rest of the changes undone.

Reply:
The only substantive change is changing `DSA_print` to `EVP_PKEY_print_private` which it wraps....

Still, easy to undo.

Reply:
Undone.